File: src/Services/BankingApp.CustomerManagement/Application/Customer/Models/CustomerData.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace BankingAppDDD.CustomerManagement.Application.Customers.Models
{
    /// <summary>
    /// 
    /// </summary>
    /// <param name="Email"></param>
    /// <param name="Name"></param>
    /// <param name="phoneno"></param>
    /// <param name="dateOfBirth"></param>
    /// <param name="customerType"></param>
    public record class CustomerData(string Email,
    string Name,
    string phoneno,
    DateOnly dateOfBirth,
    int customerType);
    
}

File: src/BankingAppDDDSpa/src/app/core/domain/customerdata.ts
import { AddressData } from "./addressdata";

export class CustomerData {
  Email!: string;
  Name!: string;
  Phoneno!: string;
  DateOfBirth!: Date;
  CustomerType!: number;
  Gender!: string;
  SSNumber!: string
  constructor(email: string, name: string, phoneno: string, dateofbirth: Date, customertype: number, gender: string, ssnumber: string){
    this.Email = email;
    this.Name = name;
    this.Phoneno = phoneno;
    this.DateOfBirth = dateofbirth;
    this.CustomerType = customertype;
    this.Gender = gender;
    this.SSNumber = ssnumber;
  }
}

export interface AddCustomerModel{
  customerData: CustomerData;
  addressData: AddressData;
}